services/core/java/com/android/server/power/PowerManagerService.java

```diff
@@ -230,6 +232,10 @@ public final class PowerManagerService extends SystemService
     // Threshold to consider proximity sensor covered
     private static final float PROXIMITY_NEAR_THRESHOLD = 5.0f;
 
+    // Smart charging: sysfs node of charger
+    private static final String POWER_INTPUT_SUSPEND_NODE =
+            "/sys/class/power_supply/battery/input_suspend";
```

Comment:
Do not hardcode this sys node. It varies on different kernels. On my k4.9 it's `/sys/class/power_supply/battery/charging_enabled`

Reply:
ok will send new commit now but can you accept this pr

Reply:
Ok so I amend this commit and force push?

Reply:
yeah, add an overlay option for sys path:
```xml
<string name="config_batteryLifeSaverSysfsNode" translatable="false">/sys/class/...</string>
```

Also, you need to add to overlay what values to write to sys node. Because `1` for `input_suspend` means _Stop charging_ but for `charging_enabled` it means _Start charging_.

Reply:
yeah, add an overlay option for sys path.
Also, you need to add to overlay what values to write to sys node. Because `1` for `input_suspend` means _Stop charging_ but for `charging_enabled` it means _Start charging_.

Reply:
ok

Reply:
Check New Commits.

Reply:
Much better. But why different grouping name? Keep `config_smartChargingBattery<something>` for everything related to smart charging feature.

Reply:
Another thing: on/off values can be strings. On Motorola Defy+ (MB526) the sysfs node is _/sys/class/power_supply/battery/status_ and it accepts _Charging/Discharging_ values instead of _1/0_; on some devices it can also be _enabled/disabled_